### PR TITLE
Product card styling bugfixes

### LINKED ISSUE #123: Product detail hidden on map view
On map view (Holiday Market) of a store with products, the second detail line of products is hidden by the Your Rewards banner. Visible if drawer is forcibly scrolled up more (bounces a bit)

scrolled to the bottom of the drawer
default

forcibly pulled up further 
scrolled up

### LINKED ISSUE #124: Product cards in See All Products are cut off on some devices
Device: Pixel 3XL on emulator (narrower screen)
Screenshot_1587413798

### LINKED ISSUE #125: Product card text overflow and spacing issues
* Need to add spacing between product cards 
* Need to truncate or somehow handle overflow text for long titles or descriptions

image

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* Several quick visual bug fixes before Nam's launch:
  * Resolved #124 product card overflow on See All Products
  * Resolved #125 product card spacing and text-overflow on the map screen -- truncated text with ellipses 
  * Resolved #123 where the products drawer cut off store details
  * Added product detail line to product details page 

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
* Make sure everything looks right!

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
* Responsive product card widths (iPhone 11 vs iPhone SE)
desk

* Truncate long product titles/details (Strawberries truncated, detail line of Trail mix truncated)
Untitled 2

* Fixed drawer height
Untitled 2

CC: @anniero98

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/product/ProductCard.js [hunk 1]
```diff
@@ -21,21 +22,29 @@ function ProductCard({
...
-          <Image
-            source={{ uri: product.imageUrl }}
-            style={{ height: 86, width: 86, borderRadius: 12 }}
-          />
-        )}
-        <Body>{product.name}</Body>
+      <ColumnContainer
+        style={
+          productsScreen
+            ? { width: (Window.width - 32 - 40) / 2 }
```
Comment (anniero98) [inline components/product/ProductCard.js:28]: responsiveness queen 👑

File: screens/map/MapScreen.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ import {
 
 const minSnapPoint = 160;
 const midSnapPoint = 325;
-const maxSnapPoint = 460;
+const maxSnapPoint = 488;
```
Comment (anniero98) [inline screens/map/MapScreen.js:29]: These numbers mean nothing to me, but 👍 
*side not: these are the types of decisions/specifics you want to document! Probably only needs an inline comment here for now

File: components/product/ProductCard.js [hunk 3]
```diff
@@ -21,21 +22,29 @@ function ProductCard({
...
+            : { width: 86 }
+        }>
+        <Image
+          source={{ uri: product.imageUrl }}
+          style={{
+            height: 86,
```
Comment (anniero98) [inline components/product/ProductCard.js:34]: I have been bewildered by the formatting changes from the last couple PRs, and just realized that eslint is picky about this too so it auto-fixes it.. 😓 We might want to remove the `--fix` from the pre-commit hook and instead just run `eslint`. I'll probably update this in my linting PRs

File: screens/map/MapScreen.js (hunk 2 above)
Comment (wangannie) [reply screens/map/MapScreen.js:29]: Will have to dig through old PRs to find the math but yes will add.

**Pull request merged**